examples/hybrid/gravitywave_parameterization/gravity_wave_parameterization.jl

```diff
@@ -67,29 +110,49 @@ function gravity_wave_tendency!(Yₜ, Y, p, t)
     # .     may be calculated
     #
 
-    # source level: get the index of the level that is closest to the source height (GFDL uses the fist level below instead)
-    source_level = argmin(
-        abs.(
-            unique(parent(Fields.coordinate_field(Y.c).z) .- gw_source_height)
-        ),
-    )
+    # source level: get the index of the level that is closest to the source height/pressure (GFDL uses the fist level below instead)
+    if model_config isa SingleColumnModel
+        source_level = similar(Fields.level(Y.c.ρ, 1))
+        Fields.bycolumn(axes(ᶜρ)) do colidx
+            parent(source_level[colidx]) .= argmin(
+                abs.(
+                    parent(Fields.coordinate_field(Y.c).z[colidx]) .-
+                    gw_source_height
+                ),
+            )[1]
+        end
+    elseif model_config isa SphericalModel
+        (; ᶜK, ᶜp) = p
+        thermo_params = CAP.thermodynamics_params(params)
+        ᶜuₕ = Y.c.uₕ
+        ᶠw = Y.f.w
+        @. ᶜK = norm_sqr(C123(ᶜuₕ) + C123(ᶜinterp(ᶠw))) / 2
+        CA.thermo_state!(Y, p, ᶜinterp)
```

Comment:
`CA.thermo_state!(Y, p, ᶜinterp)` was already been called before `gravity_wave_tendency!` is called, so I don't think this is needed

Reply:
I see. It looks like it has been called in `precomputed_quantities!`. In fact, all I need here is pressure and it looks like `ᶜp` is also updated there.